packages/web-api/src/methods.ts

```diff
@@ -364,6 +364,55 @@ export interface FilesCommentsDeleteArguments extends WebAPICallOptions, TokenOv
   id: string; // comment id
 }
 
+// either file or external_id is required
+interface FilesRemoteInfoCommonArguments extends WebAPICallOptions, TokenOverridable { channel?: string; }
+type FilesRemoteInfoArgumentsFileRequired = FilesRemoteInfoCommonArguments & { file: string };
+type FilesRemoteInfoArgumentsExternalIdRequired = FilesRemoteInfoCommonArguments & { external_id: string };
+export type FilesRemoteInfoArguments = FilesRemoteInfoArgumentsFileRequired | FilesRemoteInfoArgumentsExternalIdRequired;
```

Comment:
i understand the goal of this technique is to enforce that values fit either of the two shapes that are required. however this pattern isn't currently used anywhere else in this project, so it's worth looking at whether we want to do this or not. in other places, we just use comments to document which fields are dependent on which other fields (you can take a look at the relationship of `FilesUploadArguments` `file` and `content` properties).

one reason we haven't done this in other places where it could have been done is for simplicity for a user to read and understand the types. adding this complexity comes at a cost to the reader.

another reason is that we have a long term goal of being able to generate this file from some common metadata. techniques like this make it much more complex, maybe even impossible, for these types to come from a script to generate the types.

once you add types, you may not be able to take them away (even those that aren't exported, because they are accessible via some of the types that are exported). its based on these reasons that i recommend flattening out this type definition to just mark the properties as optional even when one of a set of them might be required, and use comments to help the user understand. we can come back and do this later, and apply whatever we decide makes sense to all the types where it makes sense at once (e.g. `FilesUploadArguments`).

Reply:
Thanks for your comment. In particular, 
>another reason is that we have a long term goal of being able to generate this file from some common metadata. techniques like this make it much more complex, maybe even impossible, for these types to come from a script to generate the types.

This part really makes sense to me. I will make these changes simpler considering your points.